zircon.jvm.examples/src/main/kotlin/org/hexworks/zircon/examples/utilities/BSPExample.kt

```diff
@@ -0,0 +1,39 @@
+package org.hexworks.zircon.examples.utilities
+
+import org.hexworks.zircon.api.Positions
+import org.hexworks.zircon.api.Sizes
+import org.hexworks.zircon.api.data.Rect
+import org.hexworks.zircon.api.util.BSPTree
+
+fun main(args : Array<String>) {
+    getMap(120, 120)
+}
+
+fun getMap(width: Int, height: Int) {
+    println("Start generating the BSP map")
+
+    val r = Rect.create(Positions.create(0, 0), Sizes.create(width, height));
+
+    val root = BSPTree(null, r)
+    root.createRooms(null)
+
+    var list = mutableListOf<BSPTree?>()
+    BSPTree.collectRooms(root, list)
```

Comment:
You can have the list parameter in `collectRooms` have a default value so you don't have to pass a parameter here.